fl4health/clients/moon_client.py

```diff
@@ -137,13 +197,34 @@ def compute_loss_and_additional_losses(
             "loss": loss,
         }
 
-        if self.contrastive_weight:
-            contrastive_loss = self.get_contrastive_loss(
-                features["features"], features["global_features"], features["old_features"]
+        if self.contrastive_weight and "old_features" in features:
+            contrastive_loss = self.contrastive_loss(
+                features["features"], features["global_features"].unsqueeze(0), features["old_features"]
             )
             total_loss += self.contrastive_weight * contrastive_loss
             additional_losses["contrastive_loss"] = contrastive_loss
 
+        elif (
```

Comment:
Perhaps I'm wrong, but I think this can just be a `if`. That is, both losses can technically be applied if the weights are non-zero. Would be overkill, but not necessarily inadmissable?